src/rustzcash.rs

```diff
@@ -191,6 +219,103 @@ pub extern "system" fn librustzcash_merkle_hash(
     write_le(tmp, &mut result[..]);
 }
 
+#[no_mangle] // ToScalar
+pub extern "system" fn librustzcash_to_scalar(
+    input: *const [c_uchar; 64],
+    result: *mut [c_uchar; 32],
+) {
+    // Should be okay, because caller is responsible for ensuring
+    // the pointer is a valid pointer to 32 bytes, and that is the
+    // size of the representation
+    let mut scalar = <Bls12 as JubjubEngine>::Fs::to_uniform(unsafe { &(&*input)[..] }).into_repr();
+
+    let result = unsafe { &mut *result };
+
+    scalar.as_mut().reverse();
```

Comment:
We should be doing some kind of `write_fs` here instead.

Reply:
So extract it into a separate function like we did for `read_fs`?

Reply:
I tried this in latest commit